kolibri/core/logger/upgrade.py

```diff
@@ -57,3 +63,36 @@ def fix_duplicated_attempt_logs():
     item and non-null masterylog_id.
     """
     consolidate_quiz_attempt_logs(AttemptLog.objects.all())
+
+
+@version_upgrade(old_version=">0.15.0,<0.18.0")
+def fix_masterylog_end_timestamps():
```

Comment:
what would be now a KDP scenario?
There we have logs that have not run `consolidate_quiz_attempt_logs` mixed with logs that have it. And KDP identifies as 0.17.3, so it'd run `fix_masterylog_end_timestamps`

So, if I understand this correctly, in KDP we should either apply both upgrades, in order, or skip both. Correct?

Reply:
We should skip both, yes. Let the Kolibris sort out their data before syncing to KDP.